File: Sem-IV/Daa-Lab/BellmanFord.py
def bellmanFord(graph,src,n):
    dist = [float('inf')]*(n+1)
    dist[src] = 0

    for i in range(n-1):
        for edge in graph:
            if dist[edge[1]] > dist[edge[0]] + edge[2]:
                dist[edge[1]] = dist[edge[0]] + edge[2]
    
    for edge in graph:
        if dist[edge[1]] > dist[edge[0]] + edge[2]:
            print("Negative cycle detected ")
            return

    print("Minimum distance from the source is ")
    for i in range(n):
        print(i,"->",dist[i])
    
n = int(input("Enter the number of vertices "))
graph = []

e = int(input("Enter the number of edges "))
print("Enter the edges ")
for i in range(e):
    x = list(map(int,input().split()))
    graph.append(x)

src = int(input("Enter the source vertex "))

bellmanFord(graph,src,n)

File: Sem-IV/DCN-Lab/Java-Programs/BellmanFord.java
// Bellman Ford Algorithm with adjacency matrix as inputs

import java.util.*;

public class BellmanFord {
    public static final int maxVal = 999;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of vertices ");
        int V = sc.nextInt();
        int[][] graph = new int[V+1][V+1];
        for (int i=0;i<=V;i++){
            for(int j=0;j<=V;j++){
                graph[i][j] = maxVal;
            }
        }
        System.out.println("Enter the number of edges ");
        int E = sc.nextInt();
        System.out.println("Enter the edges (u,v) and weight of edge,Enter 0 if there is no edge ");
        for(int i=0;i<E;i++){
            int u = sc.nextInt();
            int v = sc.nextInt();
            int wt = sc.nextInt();
            graph[u][v] = wt;
        }
        System.out.println("Enter the source ");
        int src = sc.nextInt();

        int[] dist = new int[V+1];
        Arrays.fill(dist,maxVal);
        dist[src] = 0;

        for(int i =1;i<V-1;i++){
            for(int u =1;u<=V;u++){
                for(int v=1;v<=V;v++){
                    if(dist[v] > dist[u] + graph[u][v]){
                        dist[v] = dist[u] + graph[u][v]; 
                    }
                }
            }
        }
        for(int u =1;u<=V;u++){
            for(int v=1;v<=V;v++){
                if(dist[v] > dist[u] + graph[u][v]){
                    System.out.println("Negative Cycle exists");
                    return;
                }
            }
        }

        System.out.println("Vertex\tDistance from source");
        for(int i=1;i<=V;i++){
            System.out.println(i+"\t"+dist[i]);
        }
        sc.close();   
    }
    
}
